service/service.go

```diff
@@ -402,6 +402,14 @@ func (s *Service) reportSystemMetrics() {
 }
 
 func (s *Service) newProxy(id int) (proxy.Proxy, error) {
+
+	cacheProvider := s.registry.GetCacheProvider()
+	// If there's no cache provider by the registry,
+	// then use one provided by the engine, if any.
+	if cacheProvider == nil {
```

Comment:
... respectively we do not need this if you agree that couple of cache with engine is not necessary.

Reply:
I'll just remove the engine coupling. :-) I didn't like it much anyway.